### PR TITLE
Add hood plugin to benchmark comparison

### LINKED ISSUE #25: Adds performance check to ci
We need to add a check ideally per PR in order to know if we are decreasing the benchmarks.

The easiest way could be a task to update the benchmark through the CI

### PR DESCRIPTION
**Pull request by AdrianRaFo**
Add Hood plugin https://github.com/47deg/hood to compare benchmarks to compare benchmarks for each PR and keep updated master benchmarks.
Also fix bintray settings.
Related to #42 
Fixes #25

### CODE REVIEW

**Comment by 47degdev** [comment]
***Hood benchmark comparison:***
:red_circle: `Decoding`

Benchmark | Value
|--|--|
master_benchmark | 11842.859677
helios_benchmark | 11782.244633


:red_circle: `Parsing`

Benchmark | Value
|--|--|
master_benchmark | 22679.801046
helios_benchmark | 22578.635723

**Comment by 47degdev** [comment]
***Hood benchmark comparison:***
:red_circle: `Decoding`

Benchmark | Value
|--|--|
master_benchmark | 11792.531299
helios_benchmark | 11676.193363


:red_circle: `Parsing`

Benchmark | Value
|--|--|
master_benchmark | 23179.89329
helios_benchmark | 22462.200543

**Comment by 47degdev** [comment]
***Hood benchmark comparison:***
:warning: `Decoding`

Benchmark | Value
|--|--|
master_benchmark | 12102.844117
helios_benchmark | 11909.714155


:warning: `Parsing`

Benchmark | Value
|--|--|
master_benchmark | 23014.861191
helios_benchmark | 22823.668568

**Comment by 47degdev** [comment]
***Hood benchmark comparison:***
:warning: `Decoding`

Benchmark | Value
|--|--|
master_benchmark | 11445.83832
helios_benchmark | 11405.04417


:warning: `Parsing`

Benchmark | Value
|--|--|
master_benchmark | 22040.969711
helios_benchmark | 21565.196362

File: .travis.yml [hunk 1]
```diff
@@ -5,5 +5,10 @@ jdk:
 
 script:
 - ./gradlew clean build test
-- if [ "$TRAVIS_BRANCH" = "master" ]; then ./gradlew :helios-benchmarks:updateMasterBenchmark; fi
 - ./gradlew :helios-docs:runAnk
+- ./gradlew :helios-benchmarks:executeMasterBenchmark
```
Comment (fedefernandez) [inline .travis.yml]: @AdrianRaFo IMO we should manually push the current master benchmarks to master in a different PR and then we could avoid this step. The uploaded benchmarks will be the latest master benchmarks. Does it make sense?

File: helios-benchmarks/build.gradle [hunk 2]
```diff
@@ -58,3 +60,22 @@ task executeBenchmarks {
     }
```
Comment (fedefernandez) [inline helios-benchmarks/build.gradle]: I'd store the benchmark generated by `executeHeliosBenchmark` in `build`

File: helios-benchmarks/build.gradle [hunk 3]
```diff
@@ -58,3 +60,22 @@ task executeBenchmarks {
...
+
+compareBenchmark {
+    previousBenchmarkPath = file('src/jmh/resources/master_benchmark.csv')
+    currentBenchmarkPath = [file('src/jmh/resources/helios_benchmark.csv')]
+    threshold = 600
+}
+
+compareBenchmarkCI {
+    previousBenchmarkPath = file('src/jmh/resources/master_benchmark.csv')
+    currentBenchmarkPath = [file('src/jmh/resources/helios_benchmark.csv')]
```
Comment (fedefernandez) [inline helios-benchmarks/build.gradle]:
```suggestion
    currentBenchmarkPath = [file('build/helios_benchmark.csv')]
```

File: helios-benchmarks/build.gradle [hunk 4]
```diff
@@ -35,7 +37,7 @@ jmh {
         resultsFile = file(jmhResultsFile)
 }
 
-task updateMasterBenchmark {
+task executeMasterBenchmark {
     doLast {
         jmh.resultsFile = file('src/jmh/resources/master_benchmark.csv')
```
Comment (fedefernandez) [inline helios-benchmarks/build.gradle]: It's not a big deal but the `csv` is not an internal resource for the benchmark module. I'd put that file in the root project directory, inside a folder called `benchmarks`

Comment (AdrianRaFo) [reply helios-benchmarks/build.gradle]: solved here https://github.com/47deg/helios/pull/45

File: helios-benchmarks/build.gradle (hunk 3 above)
Comment (AdrianRaFo) [reply helios-benchmarks/build.gradle]: solved here https://github.com/47deg/helios/pull/45

File: helios-benchmarks/build.gradle (hunk 2 above)
Comment (AdrianRaFo) [reply helios-benchmarks/build.gradle]: solved here https://github.com/47deg/helios/pull/45

File: .travis.yml (hunk 1 above)
Comment (AdrianRaFo) [reply .travis.yml]: solved here https://github.com/47deg/helios/pull/45

**Pull request merged**